apps/query-server/queryserver/main.py

```diff
@@ -67,29 +97,158 @@ async def get_index(
 ) -> Index:
     """Return details on the given index."""
 
-    schema = util.get_schema(sqclient, index)
+    schema = sqclient.get_opensearch_schema(index)
     return Index(index=index, index_schema=schema)
 
 
 @app.post("/v1/plan")
 async def generate_plan(query: Query) -> LogicalPlan:
     """Generate a query plan for the given query, but do not run it."""
 
-    plan = sqclient.generate_plan(query.query, query.index, util.get_schema(sqclient, query.index))
-    return plan
-
-
-@app.post("/v1/plan/run")
-async def run_plan(plan: LogicalPlan) -> SycamoreQueryResult:
-    """Run the provided query plan."""
+    if query.query is None:
+        raise ValueError("query is required")
+    if query.plan is not None:
+        raise ValueError("plan must not be specified")
 
-    return sqclient.run_plan(plan)
+    plan = sqclient.generate_plan(query.query, query.index, sqclient.get_opensearch_schema(query.index))
+    return plan
 
 
-@app.post("/v1/query")
-async def run_query(query: Query) -> QueryResult:
-    """Generate a plan for the given query, run it, and return the result."""
+def doc_to_json(doc: Document) -> Optional[dict[str, Any]]:
+    """Render a Document as a JSON object. Returns None for MetadataDocuments."""
+    NUM_TEXT_CHARS_GENERATE = 1024
+
+    if isinstance(doc, MetadataDocument):
+        return None
+
+    props_dict = {}
+    props_dict.update(doc.properties)
+    if "_schema" in props_dict:
+        del props_dict["_schema"]
+    if "_schema_class" in props_dict:
+        del props_dict["_schema_class"]
+    if "_doc_source" in props_dict:
+        del props_dict["_doc_source"]
+    props_dict["text_representation"] = (
+        doc.text_representation[:NUM_TEXT_CHARS_GENERATE] if doc.text_representation is not None else None
+    )
+    return props_dict
+
+
+async def run_query_stream(query: Query) -> EventSourceResponse:
+    """Streaming version of run_query. Returns a stream of results as they are generated."""
+
+    async def query_runner():
+        try:
+            logger.info(f"Generating plan for {query.index}: {query.query}")
+            yield {
+                "event": "status",
+                "data": "Generating plan",
+            }
+            await asyncio.sleep(0.1)
+            plan = sqclient.generate_plan(query.query, query.index, sqclient.get_opensearch_schema(query.index))
+            logger.info(f"Generated plan: {plan}")
+            # Don't want to return these through the API.
```

Comment:
I'd recommend allowlisting rather than denylisting -- the plan can be really large (I don't know if this fixes it). For the current code in the customer I explicitly only returned the nodes which was always small along with some other small debugging information.

Reply:
I can make this fix later if needed but I am being expedient for now.